cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/cloudstorage/coordinated/CassandraClusterInfoGroup.java

```diff
@@ -266,4 +276,14 @@ private RuntimeException toRuntimeException(ClusterInfo clusterInfo, Throwable c
         LOGGER.error("Failed to perform action on cluster. cluster={}", clusterInfo.clusterId(), cause);
         return new RuntimeException("Failed to perform action on cluster: " + clusterInfo.clusterId(), cause);
     }
+
+    private static void validateClusterInfoList(List<ClusterInfo> clusterInfos)
+    {
+        Preconditions.checkArgument(clusterInfos != null && !clusterInfos.isEmpty(),
+                                    "clusterInfos cannot be null or empty");
+        for (ClusterInfo ci : clusterInfos)
+        {
+            Preconditions.checkArgument(ci.clusterId() != null, "Found clusterInfo without clusterId defined");
```

Comment:
there's no easy way to identify which ClusterInfo is missing the clusterId, so it might be hard to debug.

Reply:
Good call. I can move the validation to the location where the original input is accessible.